h2o-core/src/test/java/water/fvec/VecTest.java

```diff
@@ -197,26 +197,16 @@ public void map(Chunk c) {
   }
 
   @Test public void testChunkIdxWithDeletedVec() {
+    Key<Vec> k = null;
     try {
       Scope.enter();
       Vec v = Scope.track(makeConN((long) 1e6, 16));
-      new MRTask() {
-        @Override
-        protected void setupLocal() {
-          Vec v = _fr.anyVec();
-          if (v != null) {
-            v.remove(); // evil happens here
-          }
-        }
-        @Override
-        public void map(Chunk c) {
-          if (c.vec().get_type() != Vec.T_NUM)
-            throw new IllegalStateException("Expected a numeric Vec");
-        }
-      }.doAll(v);
+      k = v._key;
+      v.remove();
+      v.chunkForChunkIdx(0);
     } catch (Exception e) {
       String msg = e.getMessage();
-      if (msg == null || !msg.contains("Missing chunk") || !msg.contains("is not in DKV"))
```

Comment:
maybe checking `e.toString()` instead of `e.getMessage()` would work here?

Reply:
Can you please explain what issue it addresses?

Reply:
I thought that as DistributedException may have been wrapping the original exception without message using basic constructor `public DistributedException(Throwable cause)`, then your test would always fail.

Reply:
for example in `RPC` and `MRTask`:
```
        throw (re instanceof DistributedException)?new DistributedException(re.getMessage(),re.getCause()):new DistributedException(re);
```
for some (unknown) reason, the message is propagated only if the cause exception is already a `DistributedException`

Reply:
my mistake, didn't see the full logic: by default it's prefixing the cause message...

Reply:
Okay, I see - that would be for the original code.

Current code won't involve distributed execution - I will always get exactly the exception I am throwing.

The issue with the previous one was not wrapping into DistributedException, it was an NPE in the initialization of the MRTask on the remote nodes.

Reply:
yes, I'm using split view (there, it is clear it was a comment on original test).
with unified view, my comment may have been confusing